src/components/main.tsx

```diff
@@ -1,24 +1,26 @@
 import { useState } from 'react';
 import { HashRouter as Router, Route, Switch } from 'react-router-dom';
 import NavBar from './NavBar/NavBar';
-import { routes } from '../assets/routes'
+import { routes } from '../assets/routes';
 
 const Main = () => {
   const [minimizeSideBar, setMinimizeSideBar] = useState(true);
-  const [mainId, setMainId] = useState("main-max");
+  const [mainWidthClass, setMainWidthClass] = useState('main-max');
 
   const toggleMainWidth = () => {
     setMinimizeSideBar(!minimizeSideBar);
-    setMainId(minimizeSideBar ? "main-min" : "main-max");
+    setMainWidthClass(minimizeSideBar ? 'main-min' : 'main-max');
   };
 
   return (
     <Router>
       <NavBar toggleMainWidth={toggleMainWidth} mini={minimizeSideBar} />
-      <div className="content" id={mainId}>
+      <div className={ ["content",mainWidthClass].join(' ') } >
         <Switch>
           {routes.map((route) => (
-            <Route key={route.link} path={route.link} component={route.comp}/>
+            <Route key={route.link}
+                   exact path={route.link} 
```

Comment:
Not quite what I meant. In this way, every one of your routes will be exact- which is not that optimal.
There are a couple of possible solutions:
1. You could take the `home` route outside of the array as an "exception", and write this one "hard-coded". The problem is... you know... having something hard-coded.
2. You could add another parameter, "exact" to `routes`, and then write `exact={route.exact}`. The problem here is that you'll have to write `exact: false` in each one...

I tend to go more with the first solution, but it's up to you. In anyway, please change it.